library/network/src/lib/y2firewall/firewalld/relations.rb

```diff
@@ -183,6 +184,7 @@ def has_many(*relations, scope: nil, cache: false) # rubocop:disable Style/Predi
           attr_reader relation
 
           define_method "#{relation}=" do |item|
+            return item if public_send(relation) == item
```

Comment:
It could be improved, doing a sort of the items but will need other checks, so not added by now.